scripts/split_eafs.py

```diff
@@ -0,0 +1,152 @@
+args = parser.parse_args()
+try:
+    input_dir = args.input_dir
+    tier = args.tier
+    silence_marker = args.silence_marker
+    silence_tier = args.silence_tier
+    output_audio_dir = args.output_audio_dir
+    output_text_dir = args.output_text_dir
+    # output json file name is passed in with path attached
+    output_json = args.output_json
+except Exception:
+    parser.print_help()
+    sys.exit(0)
+
+
+if not os.path.exists(output_audio_dir):
+    os.makedirs(output_audio_dir)
+if not os.path.exists(output_text_dir):
+    os.makedirs(output_text_dir)
+
+
+def split_audio_by_start_end(input_audio, start, end, fname, ext):
+    output = input_audio[start:end]
+    output.export(os.path.join(output_audio_dir, fname + ext), format=ext[1:])
+
+
+def write_text(annotation, fname, ext):
+    f = open(os.path.join(output_text_dir, fname + ext), 'w')
+    f.write(annotation)
+    f.close()
+
+
+def write_json(annotations_data):
+    with open(output_json, 'w') as outfile:
+        json.dump(annotations_data, outfile, indent=4, separators=(',', ': '), sort_keys=False)
+
+
+def read_eaf(ie):
+
+    input_eaf = Eaf(ie)
+
+    # Check if the tiers we have been given exist
+    tier_names = input_eaf.get_tier_names()
+    if tier not in tier_names:
+        print('missing tier: ' + tier, file=sys.stderr)
+        return False
+    if silence_tier not in tier_names:
+        print('missing silence tier: ' + silence_tier, file=sys.stderr)
+
+    # get the input audio file
+    inDir, name = os.path.split(ie)
+    basename, ext = os.path.splitext(name)
+    ia = os.path.join(inDir, basename + ".wav")
+    input_audio = AudioSegment.from_wav(ia)
+
+    # We can pass in an arg for a ref tier that has silence labels
+    check_silence_ref_tier = False
+    if silence_tier in tier_names:
+        silence_tier_info = input_eaf.get_parameters_for_tier(silence_tier)
+        if silence_tier_info.get("PARENT_REF") == tier:
+            check_silence_ref_tier = True
+
+    # Get annotation values, start and end times, and speaker id
+    annotations = sorted(input_eaf.get_annotation_data_for_tier(tier))
+    params = input_eaf.get_parameters_for_tier(tier)
+    if 'PARTICIPANT' in params:
+        speaker_id = params['PARTICIPANT']
+
+    annotations_data = []
+    i = 0
+    for ann in annotations:
+        skip = False
+        ref_annotation = []
+        start = ann[0]
+        end = ann[1]
+        annotation = ann[2]
+
+        # Check for annotations labelled with a particular symbol on the main tier
+        if annotation == silence_marker:
+            skip = True
+
+        # Check for existence of an annotation in ref tier to silence
+        # Annotation value doesn't matter
+        if check_silence_ref_tier:
+            ref_annotation = input_eaf.get_ref_annotation_at_time(silence_tier, start)
+            if len(ref_annotation) is True:
```

Comment:
This line will never execute. What is the check intending to do here?

Reply:
Don't worry too much about being new to Python we all started somewhere :) The formatting in my comment isn't as clear as I'd like, specifically it's the conditional on line 112 that will never execute since `len(ref_annotation)` can never be `True`.

if you simply want to skip if `ref_annotation` has some items in it I'd suggest something along the lines of:
```python
if check_silence_ref_tier:
    ref_annotation = input_eaf.get_ref_annotation_at_time(silence_tier, start)
    if len(ref_annotation):
        skip = True
``` 
Or perhaps remove the `ref_annotation` variable entirely (as it appears to never be used past this point) with something like:

```python
if check_silence_ref_tier and len(input_eaf.get_ref_annotation_at_time(silence_tier, start)):
    skip = True
```

Reply:
Ah, I see. Yes, your second method is great. Can you update or should I submit a new PR?

Reply:
The workflow here is to make a pull request for changes so please open another PR. I'm happy to make a comprehensive code review with detailed feedback on Python related matters in that PR if you want.

Reply:
Thank you, I'll do it in the morning. I'd appreciate feedback if you have time. Keen to learn.

Reply:
Any updates on this?

Reply:
Just submitted PR.. Last few days were too hectic to give full attention to this.